### PR TITLE
Handle missing test cases in Python grader

### PR DESCRIPTION
**Pull request by nwalters512**
If an instructor doesn't name their test case functions correctly (that is, they don't have a `test_` prefix), `unittest` won't be able to discover them. If there aren't any tests, there won't be any points, so we'll end up in a place where we try to compute `0 / 0`, which will of course fail. This PR special-cases that edge case so that we report a score of 0 if there weren't any tests.

I also added a user-facing message if no tests were found. Otherwise, `<pl-external-grader-results>` would just render "Score: 0%", which doesn't make it entirely clear that there weren't any tests to run.

Arguably we could consider it to be a "format error" if there aren't any tests and thus flag the submission as ungradable, but IMO that's overkill.

### CODE REVIEW

**Review by mylesw27 — APPROVED** [review]

File: graders/python/python_autograder/pl_main.py [hunk 1]
```diff
@@ -115,6 +116,13 @@ def add_files(results):
         if len(format_errors) > 0:
             grading_result["format_errors"] = format_errors
 
+        # Instructors may have named their tests incorrectly or somehow misconfigured things.
+        # Help point them in the right direction.
+        if len(results) == 0:
+            grading_result["message"] = (
+                "No tests were found. Please check your test cases."
```
Comment (jonatanschroeder) [inline graders/python/python_autograder/pl_main.py:123]: I don't think this should be an error. Cases like compilation errors often happen before tests are created, and in that case the message and/or output will have relevant content without any tests present. I think a generic message is fine, but only if both message and output are blank, and should not suggest an action. Something like "No tests were found" is enough.

Comment (jonatanschroeder) [reply graders/python/python_autograder/pl_main.py:123]: I just tested the autograder with the changes above. If I submit an answer with compilation errors, the error message above is added, even though the question has tests. The tests were not reached because of the compilation error, so this message was actually incorrect/misleading.

Upon further review of this code, I believe there are better ways to identify if a problem is indeed a case where no tests were found, such as checking if `gradable` is `True` or keeping track of the `TestLoader` results in line 63, calling [`countTestCases()`](https://docs.python.org/3/library/unittest.html#unittest.TestSuite.countTestCases).

Comment (nwalters512) [reply graders/python/python_autograder/pl_main.py]: I changed to using `suite.countTestCases() > 0` in aaeaae7. How does that look?

File: apps/prairielearn/elements/pl-external-grader-results/pl-external-grader-results.mustache
Comment (nwalters512) [inline apps/prairielearn/elements/pl-external-grader-results/pl-external-grader-results.mustache:1]: I noticed that the spacing was a little funky in this element, particularly when it's followed by another element (there was zero vertical spacing after the grader results). Let me know if you want this pulled into another PR.

File: graders/python/python_autograder/pl_main.py [hunk 2]
```diff
@@ -91,6 +93,7 @@ def add_files(results):
         # Compile total number of points
         max_points = test_case.get_total_points()
         earned_points = sum([test["points"] for test in results])
+        score = 0 if max_points == 0 else float(earned_points) / float(max_points)
```
Comment (jonatanschroeder) [inline graders/python/python_autograder/pl_main.py:96]: Nit:
```suggestion
        score = 0 if float(max_points) == 0 else float(earned_points) / float(max_points)
```

**Review by jonatanschroeder — APPROVED** [review]

File: graders/python/python_autograder/pl_main.py (hunk 2 above)
Comment (eliotwrobson) [reply graders/python/python_autograder/pl_main.py:96]: If you do this, make sure to use `math.isclose` instead of `==`.

Comment (nwalters512) [reply graders/python/python_autograder/pl_main.py:96]: Oh? Say more, I'm curious when a float value would not equal zero but also cause a `ZeroDivisionError`.

Comment (eliotwrobson) [reply graders/python/python_autograder/pl_main.py:96]: It won't throw the exception, but a good general rule to always avoid comparing floats using `==`. It probably doesn't matter in this case, though it might if `max_points` is a computed value.

Comment (nwalters512) [reply graders/python/python_autograder/pl_main.py:96]: IMO we don't care if `max_points` is _almost_ zero, which could (in some very strange situation) be valid. Only if it's actually zero, in which case we need to avoid a divide-by-zero situation. Does that sound right?

Comment (eliotwrobson) [reply graders/python/python_autograder/pl_main.py]: That makes sense 👍
